### PR TITLE
feat(mnq): add a create-context nats custom command

### PR DESCRIPTION
**Pull request by jremy42**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/master/CHANGELOG.md):

```release-note

```

### CODE REVIEW

File: internal/namespaces/mnq/v1beta1/custom_nats.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+Credentials and context file are saved in your nats context folder with 0600 permissions`,
+		Examples: []*core.Example{
+			{
+				Short: "Create a context in your nats server",
+				Raw:   `scw mnq nats create-context <nats-account-id> credentials-name=<credential-name> region=fr-par`,
+			},
+		},
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:  "nats-account-id",
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats.go:57]: Should be positional

File: internal/namespaces/mnq/v1beta1/custom_nats.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+				Short: "ID of the NATS account",
+			},
+			{
+				Name:  "name",
+				Short: "Name of the saved context, defaults to account name",
+			},
+			{
+				Name:    "credentials-name",
+				Short:   "Name of the created credentials",
+				Default: core.RandomValueGenerator("mnq"),
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats.go:67]: Could you make sure that both files have matching names.

File: internal/namespaces/mnq/v1beta1/custom_nats_helpers.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	}
+	_, _ = interactive.Println("Credentials for " + creds.Name + " have been successfully written to " + credsPath)
+	return credsPath, nil
+}
+
+func getNATSContextDir(ctx context.Context) (string, error) {
+	xdgConfigHome := core.ExtractEnv(ctx, "XDG_CONFIG_HOME")
+	if xdgConfigHome == "" {
+		homeDir := core.ExtractEnv(ctx, "HOME")
+		fmt.Println("HomeDir in getNatsContextDir", homeDir)
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats_helpers.go:96]: Should be interactive or deleted

File: internal/namespaces/mnq/v1beta1/custom_nats_helpers.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	}
+	if fileExists(contextPath) {
+		overWrite, err := promptOverWriteFile(ctx, contextPath)
+		if err != nil {
+			return "", wrapError(err, "Failed to write context", natsAccount.Name, contextPath)
+		}
+		if !overWrite {
+			return "", wrapError(nil, "File Already exist", natsAccount.Name, contextPath)
+		} else {
+			err := os.Remove(credsPath)
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats_helpers.go:52]: Not useful if file is going to be written

File: internal/namespaces/mnq/v1beta1/custom_nats_helpers.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+		}
+	}
+	if err := os.WriteFile(contextPath, natsContextFrom(natsAccount, credsPath), 0600); err != nil {
+		return "", wrapError(err, "Failed to write context", natsAccount.Name, contextPath)
+	}
+	_, _ = interactive.Println("Context file for " + natsAccount.Name + " has been successfully written to " + contextPath)
+	return contextPath, nil
+}
+
+func writeCredsFile(ctx context.Context, natsContextDir string, creds *mnq.NatsCredentials) (string, error) {
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats_helpers.go:65]: Same logic than with writeCredsFile. Try to factorize this.

Comment (remyleone) [reply internal/namespaces/mnq/v1beta1/custom_nats_helpers.go:65]: Maybe there is existing function that do that

File: internal/namespaces/mnq/v1beta1/custom_nats_test.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/scaleway/scaleway-cli/v2/internal/core"
+)
+
+func Test_CreateContext(t *testing.T) {
+	t.Run("Simple", core.Test(&core.TestConfig{
+		Commands:   GetCommands(),
+		BeforeFunc: createNATSAccount("NATS"),
+		Cmd:        "scw mnq nats create-context nats-account-id={{ .NATS.ID }}",
+		Check: core.TestCheckCombine(
+			//core.TestCheckGolden(), > As the filenames change, the golden tests do not pass.
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats_test.go:17]: Try to fix the name to have the golden identical

File: internal/namespaces/mnq/v1beta1/custom_nats_test.go [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+			return nil
+		},
+		),
+	}))
+}
+
+func Test_CreateContextWithWrongId(t *testing.T) {
+	t.Run("Wrong Account ID", core.Test(&core.TestConfig{
+		Commands: GetCommands(),
+		Cmd:      "scw mnq nats create-context nats-account-id=Wrong-id",
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/custom_nats_test.go:50]: Should be changed to positional

File: internal/namespaces/mnq/v1beta1/helpers_test.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+func deleteNATSAccount(metaKey string) core.AfterFunc {
+	return core.ExecAfterCmd("scw mnq nats delete-account {{ ." + metaKey + ".ID }}")
+}
+
+func deleteFiles() core.AfterFunc {
+	return nil
+}
+
+func extractFileName(path string, ext string) string {
```
Comment (remyleone) [inline internal/namespaces/mnq/v1beta1/helpers_test.go:25]: Could filepath.Base() work ? Could you change the name?

File: internal/namespaces/mnq/v1beta1/custom_nats_test.go (hunk 6 above)
Comment (Codelax) [reply internal/namespaces/mnq/v1beta1/custom_nats_test.go:17]: core provides `core.TestCheckGoldenAndReplacePatterns` to check goldens with regex to circumvent this kind of issue

File: internal/namespaces/mnq/v1beta1/custom_nats.go [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+type CreateContextRequest struct {
+	NatsAccountID   string
+	ContextName     string
+	CredentialsName string
+	Region          scw.Region
+}
+
+func aliasCreateContextCommand() *core.Command {
```
Comment (Codelax) [inline internal/namespaces/mnq/v1beta1/custom_nats.go:39]: Change the function name, this one is remaining from when I copied alias command

**Review by remyleone — APPROVED** [review]

**Pull request merged**